### ISSUE TITLE
applying to a round is not working

### ISSUE THREAD

**Issue by boudra** [gitcoinco/grants-stack#3169]
this is applying to a round in avalanche using an optimism project

https://builder-git-allo-migration-grants-stack.vercel.app/#/chains/43114/rounds/1

1. ensure the UI doesn't hang and handles errors properly
2. fix the actual issue

<img width="1512" alt="Screenshot 2024-03-21 at 14 34 46" src="https://github.com/gitcoinco/grants-stack/assets/711886/042d58a3-271c-41fa-8bd7-7d4e2f5bd4d8">

**Comment by thelostone-mc** [comment]
Fixed at 70c3c87a

**Comment by thelostone-mc** [comment]
So here is what we found out. 
The anchor address on optimism is different from the ones on the other chains for a given project despite having the same project id.
This is something along the lines what we were concerned for zk-era 

We created a project on optimism and created linked profiles on scrolll and avalanche. Then we checked the anchor address and notices that while all projects have the same projectId, The anchor address for scroll and avalanche were same but different from what optimism had.

```
{
  projects(condition: {id: "0x5f7b5be0f364eab38af5468c2dd16c7926101d51752ce1ef6a66f57f7aefb07b", tags: "allo-v2"}) {
    id
    chainId
    anchorAddress
    name
    nonce
  }
}
```

The fix: we would have to update builder to ensure we pick the right projects anchor address and do away with the assumption that the anchor address is the same. 

Why does this happen :
https://github.com/ethereum-optimism/contracts/issues/306
OVM bytecode is different from EVM bytecode
Create2 addresses on Optimism would be different from Create2 addresses in Ethereum.

**Downside: **
- No downside apart from projects having different anchor address
 - Extra work to update builder to always use the right anchor (low effort)

**Upside**
- The solution we put in here on builder will unblock era
